### PR TITLE
v1

### PR DESCRIPTION
**Pull request by mul53**

### CODE REVIEW

File: src/graphql/index.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import fetch from 'cross-fetch'
```
Comment (leonprou) [inline src/graphql/index.ts:1]: 👍

File: src/staking/Staking.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import Web3 from 'web3'
+
+export default abstract class Staking {
```
Comment (leonprou) [inline src/staking/Staking.ts:3]: what you think about some renaming here?
the class name is confusing

I suggest `RewardProgram`

File: src/staking/Staking.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  protected readonly web3: Web3
+
+  constructor (protected readonly stakingAddress: string, provider: any) {
```
Comment (leonprou) [inline src/staking/Staking.ts:6]: is `stakingAddress` exposed from the outside?

File: src/constants/abi/StakingSingleReward.json [hunk 4]
```diff
@@ -0,0 +1,319 @@
+[
+	{
+		"constant": false,
```
Comment (leonprou) [inline src/constants/abi/StakingSingleReward.json:3]: the convention is to call the ABI as the contract name

**Review by leonprou — CHANGES_REQUESTED** [review]
Looks good 👍 
please see comments

File: src/staking/Staking.ts (hunk 3 above)
Comment (mul53) [reply src/staking/Staking.ts:6]: changed it to expose

File: src/constants/abi/StakingSingleReward.json (hunk 4 above)
Comment (mul53) [reply src/constants/abi/StakingSingleReward.json:3]: yes that's the convention

File: src/staking/Staking.ts (hunk 2 above)
Comment (mul53) [reply src/staking/Staking.ts:3]: okay i have changed the name to RewardProgram

**Review by leonprou — APPROVED** [review]

**Pull request merged**